internal/pkg/template/templates/environment/manifest.yml

```
@@ -0,0 +1,77 @@
+# The manifest for the "{{.Name}}" environment.
+# Read the full specification for the "{{.Type}}" type at:
+#  https://aws.github.io/copilot-cli/docs/manifest/environment/
+
+# Your environment name will be used in naming your resources like VPC, cluster, etc.
+Name: {{.Name}}
+Type: {{.Type}}
+
+# Import your own VPC and subnets or configure how they should be created.
+{{- if .Network.VPC.IsEmpty}}
+# network:
+#   vpc:
+#     id:
+{{- else}}
+network:
+  {{- $vpc := .Network.VPC}}
+  vpc:
+    {{- if $vpc.ID}}
+    id: {{$vpc.ID}}
+    {{- else if $vpc.CIDR }}
+    cidr: {{$vpc.CIDR }}
+    {{- end }}
+    {{- if not $vpc.Subnets.IsEmpty}}{{- $subnets := $vpc.Subnets}}
+    subnets:
+      {{- if $subnets.Public}}
+      public:
+        {{- range $subnet := $subnets.Public}}
+        - {{- if $subnet.SubnetID}} id: {{$subnet.SubnetID}} 
+          {{- else if $subnet.CIDR}} cidr: {{$subnet.CIDR}}
+          {{if $subnet.AZ -}} az: {{$subnet.AZ}} {{- end}}
```

is this  `{{- end}}` for `subnet.AZ`? I think I'm confused why there isn't an `{{- end}}` after the {{- else if}}

---

Why did you prefer removing the trailing spaces after, rather than before?
```suggestion
          {{- if $subnet.AZ}} az: {{$subnet.AZ}} {{- end}}
```

---

Because that trim the `\n` (also a whitespace) from the last line as well 😢